crates/junction-api/src/kube/http.rs

```diff
@@ -33,6 +33,17 @@ macro_rules! vec_from_gateway {
     };
 }
 
+macro_rules! vec_from_gateway_p {
```

Comment:
Let's get slightly fancy and combine this with `vec_from_gateway`. Macros can have overloaded arg lists even though functions can't. We already do this with `vec_to_gateway` if you want another example.

```rust
macro_rules! vec_from_gateway {
    ($t:ty, $opt_vec:expr, $field_name:literal) => {
        $opt_vec
            .iter()
            .flatten()
            .enumerate()
            .map(|(i, e)| <$t>::from_gateway(e).with_field_index($field_name, i))
            .collect::<Result<Vec<_>, _>>()
    };
    ($t:ty, $opt_vec:expr, $field_name:literal, $param:expr) => {
        $opt_vec
            .iter()
            .flatten()
            .enumerate()
            .map(|(i, e)| <$t>::from_gateway($param, e).with_field_index($field_name, i))
            .collect::<Result<Vec<_>, _>>()
    };
}
```